### PR TITLE
CI: replace compiletest-rs with trybuild

### PR DESCRIPTION
**Pull request by japaric**
We use compiletest to run compile-fail tests but compiletest depends on compiler
internals so it breaks every now and then and requires nightly. With trybuild we
can also run compile-fail tests but it works on stable and it already has
reached version 1.0

### CODE REVIEW

File: ui/single/task-priority-too-high.stderr [hunk 1]
```diff
@@ -1,9 +1,13 @@
+warning: unused import: `rtfm::app`
+ --> $DIR/task-priority-too-high.rs:3:5
+  |
+3 | use rtfm::app;
```
Comment (jonas-schievink) [inline ui/single/task-priority-too-high.stderr]: Uh, spurious warning? Is this a known rustc bug?

File: ci/script.sh [hunk 2]
```diff
@@ -37,10 +37,10 @@ main() {
     mkdir -p ci/builds
 
     if [ $T = x86_64-unknown-linux-gnu ]; then
-        if [ $TRAVIS_RUST_VERSION = nightly ]; then
-            # compile-fail tests
-            cargo test --test single --target $T
+        # compile-fail tests
+        cargo test --test single --target $T
```
Comment (jonas-schievink) [inline ci/script.sh]: The compiler output might change between versions, so running this on all channels might cause issues later on

Comment (japaric) [reply ci/script.sh]: good point. I have changed the condition to test only when `rustc --version == $MSRV`. $MSRV remains constant for a whole major version of rtfm (e.g. 1.36.0 for v0.5.x)

File: ui/single/task-priority-too-high.stderr (hunk 1 above)
Comment (japaric) [reply ui/single/task-priority-too-high.stderr]: The warning it's also present on 1.36.0. Perhaps this has been there for a while but compiletest ignores warnings by default?

Comment (jonas-schievink) [reply ui/single/task-priority-too-high.stderr]: Yeah, that's probably what's happening

**Pull request merged**